scripts/mbedtls_dev/bignum_core.py

```diff
@@ -818,6 +820,204 @@ def mpi_modmul_case_generate() -> None:
+            self.case_description = "{} {} {} mod {}".format(
+                self.value_description(self.int_a, self.desc_a),
+                self.symbol,
+                self.value_description(self.int_b, self.desc_b),
+                self.value_description(self.int_m, self.desc_m)
+            )
+        return super().description()
+
+    @classmethod
+    def get_value_triplets(cls) -> Iterator[
+            Tuple[Tuple[str, str], Tuple[str, str], Tuple[str, str]]]:
+        for a in cls.input_1st_operands:
+            for b in cls.input_2nd_operands:
+                for m in cls.input_moduli:
+                    yield (a, b, m)
+
+    def is_valid(self) -> bool:
+        if self.int_a >= self.int_m or self.int_b >= self.int_m:
+            return False
+        return True
+
+    @classmethod
+    def generate_function_tests(cls) -> Iterator[test_case.TestCase]:
+        """Generator to yield test cases.
+
+        All combination of first and second operands with all moduli are
+        generated.
+
+        If a combination is not a valid test case or we don't want to test it
+        for some reason, then the constructor of the subclass should raise a
+        `ValueError`.
+        """
+        yield from (valid_test_object.create_test_case()
+                    for valid_test_object in filter(
+                        lambda test_object: test_object.is_valid(),
+                        (cls(a, b, m) for a, b, m in cls.get_value_triplets())
+                        ))
+
+# Generated by bignum_common.gen_safe_prime(192,1)
+SAFE_PRIME_192_BIT_SEED_1 = "d1c127a667786703830500038ebaef20e5a3e2dc378fb75b"
+
+# First number generated by random.getrandbits(192) - seed(2,2), not a prime
+RANDOM_192_BIT_SEED_2_NO1 = "177219d30e7a269fd95bafc8f2a4d27bdcf4bb99f4bea973"
+
+# Second number generated by random.getrandbits(192) - seed(2,2), not a prime
+RANDOM_192_BIT_SEED_2_NO2 = "cf1822ffbc6887782b491044d5e341245c6e433715ba2bdd"
+
+# Third number generated by random.getrandbits(192) - seed(2,2), not a prime
+RANDOM_192_BIT_SEED_2_NO3 = "3653f8dd9b1f282e4067c3584ee207f8da94e3e8ab73738f"
+
+# Fourth number generated by random.getrandbits(192) - seed(2,2), not a prime
+RANDOM_192_BIT_SEED_2_NO4 = "ffed9235288bc781ae66267594c9c9500925e4749b575bd1"
+
+# Ninth number generated by random.getrandbits(192) - seed(2,2), not a prime
+RANDOM_192_BIT_SEED_2_NO9 = "2a1be9cd8697bbd0e2520e33e44c50556c71c4a66148a86f"
+
+# Generated by bignum_common.gen_safe_prime(1024,3)
+SAFE_PRIME_1024_BIT_SEED_3 = ("c93ba7ec74d96f411ba008bdb78e63ff11bb5df46a51e16b"
+                              "2c9d156f8e4e18abf5e052cb01f47d0d1925a77f60991577"
+                              "e128fb6f52f34a27950a594baadd3d8057abeb222cf3cca9"
+                              "62db16abf79f2ada5bd29ab2f51244bf295eff9f6aaba130"
+                              "2efc449b128be75eeaca04bc3c1a155d11d14e8be32a2c82"
+                              "87b3996cf6ad5223")
```

Comment:
Since these have `1024` bits, do we need to worry about `MPI_MAX_BITS_LARGER_THAN_792`?

Reply:
Functions for the old interface are using `MBEDTLS_MPI_MAX_SIZE` to determine the size of some stack allocated arrays. As long as we don't rely on `MBEDTLS_MPI_MAX_SIZE`, we don't need `MPI_MAX_BITS_LARGER_THAN_792`.